luminance/examples/12-displacement-map.rs

```diff
@@ -0,0 +1,171 @@
+//! Press <escape> to quit or close the window.
+//!
+//! > Note: for this example, it is recommended to compile with --release to speed up image loading.
+//!
+//! https://docs.rs/luminance
+
+use luminance::blending::{Equation, Factor};
+use luminance::context::GraphicsContext as _;
+use luminance::pipeline::BoundTexture;
+use luminance::pixel::{NormRGB8UI, NormUnsigned};
+use luminance::render_state::RenderState;
+use luminance::shader::program::{Program, Uniform};
+use luminance::tess::{Mode, TessBuilder};
+use luminance::texture::{Dim2, Flat, GenMipmaps, Sampler, Texture};
+use luminance_derive::UniformInterface;
+use luminance_glfw::{Action, GlfwSurface, Key, Surface, WindowDim, WindowEvent, WindowOpt};
+use std::env;
+use std::time::Instant;
+
+const VS: &str = include_str!("./12-displacement-map-resources/displacement-map-vs.glsl");
+const FS: &str = include_str!("./12-displacement-map-resources/displacement-map-fs.glsl");
+
+#[derive(UniformInterface)]
+struct ShaderInterface {
+    image: Uniform<&'static BoundTexture<'static, Flat, Dim2, NormUnsigned>>,
+    displacement_map_1: Uniform<&'static BoundTexture<'static, Flat, Dim2, NormUnsigned>>,
+    displacement_map_2: Uniform<&'static BoundTexture<'static, Flat, Dim2, NormUnsigned>>,
+    displacement_scale: Uniform<f32>,
+    time: Uniform<f32>,
+    window_dimensions: Uniform<[f32; 2]>,
+}
+
+fn main() {
+    let texture_path = env::args()
+        .skip(1)
+        .next()
+        .expect("Please provide an image path for the core texture");
+    let texture_image = image::open(&texture_path)
+        .expect("Could not load image from path")
+        .flipv()
+        .to_rgb();
+    let (width, height) = texture_image.dimensions();
+
+    let displacement_map_1 = image::load_from_memory_with_format(
+        include_bytes!("./12-displacement-map-resources/displacement_1.png"),
+        image::ImageFormat::PNG,
+    )
+    .expect("Could not load displacement map")
+    .to_rgb();
+
+    let displacement_map_2 = image::load_from_memory_with_format(
+        include_bytes!("./12-displacement-map-resources/displacement_2.png"),
+        image::ImageFormat::PNG,
+    )
+    .expect("Could not load displacement map")
+    .to_rgb();
+
+    let mut surface = GlfwSurface::new(
+        WindowDim::Windowed(width, height),
+        "texture_demo",
+        WindowOpt::default(),
+    )
+    .expect("Could not create GLFW surface");
+
+    let texels = texture_image.into_raw();
+    let tex = Texture::<Flat, Dim2, NormRGB8UI>::new(
+        &mut surface,
+        [width, height],
+        0,
+        Sampler::default(),
+    )
+    .expect("Could not create luminance texture");
+    tex.upload_raw(GenMipmaps::No, &texels).unwrap();
+
+    let texels = displacement_map_1.into_raw();
+    let displacement_tex_1 =
+        Texture::<Flat, Dim2, NormRGB8UI>::new(&mut surface, [128, 128], 0, Sampler::default())
```

Comment:
Now I’m thinking that it could be great to add a `Texture` function to create it from a slice of data, but I’m not sure how to do it. Just an idea, your code is fine.